apps/frontend/src/components/cards/controltable/ControlTable.vue

```diff
@@ -219,6 +233,20 @@ export default class ControlTable extends Vue {
     }
   }
 
+  selectAllControls() {
+    if (this.allViewedValue) {
+      this.viewedControlIds = [];
+      this.allViewed = Trinary.Off;
+    } else {
+      this.items.forEach((item) => {
+        if (!this.viewedControlIds.includes(item.control.data.id)) {
```

Comment:
if all you need to do is add unique values to the list, then make 'viewedControlIds' a set so that that process is automatically handled by the data structure